pkg/machineproviders/providers/openshift/machine/v1beta1/failuredomain/set_test.go

```diff
@@ -65,90 +65,153 @@ var _ = Describe("Set suite", func() {
 
 			var set *Set
 
-			BeforeEach(func() {
-				By("creating a new set with 2 failure domains")
-				set = NewSet(usEast1aFailureDomain, usEast1cFailureDomain)
-			})
-
-			It("should have the initial failure domains", func() {
-				Expect(set.Has(usEast1aFailureDomain)).To(BeTrue(), "Set should contain the us-east-1a failure domain")
-				Expect(set.Has(usEast1cFailureDomain)).To(BeTrue(), "Set should contain the us-east-1c failure domain")
-			})
+			Context("when having an initial duplicates in the set", func() {
```

Comment:
Nit, wording suggestion
```suggestion
			Context("with duplicate failure domains in the initial set", func() {
```